test/RunSpec.hs

```diff
@@ -93,6 +94,28 @@ spec = do
         rmDir "test/integration/custom-package-conf/packages/"
         rmDir "test/integration/custom-package-conf/foo/dist/"
 
+    it "respects HASKELL_PACKAGE_SANDBOXES" $ do
+      withCurrentDirectory "test/integration/custom-package-conf/foo" $ do
+        ExitSuccess <- rawSystem "ghc-pkg" ["init", "../packages"]
+        ExitSuccess <- rawSystem "ghc-pkg" ["init", "../packages-extra"]
+        ExitSuccess <- rawSystem "cabal" ["configure", "--disable-optimization", "--disable-library-profiling", "--package-db=../packages"]
+        ExitSuccess <- rawSystem "cabal" ["build"]
+        ExitSuccess <- rawSystem "cabal" ["register", "--inplace"]
+        return ()
+
+      let sandboxes = intercalate [searchPathSeparator]
+            [ "test/integration/custom-package-conf/packages-extra"
+            , "test/integration/custom-package-conf/packages"
+            ]
+      withEnv "HASKELL_PACKAGE_SANDBOXES" sandboxes $ do
+        hCapture_ [stderr] (doctest ["test/integration/custom-package-conf/Bar.hs"])
+          `shouldReturn` "Examples: 2  Tried: 2  Errors: 0  Failures: 0\n"
+
+      `E.finally` do
+        rmDir "test/integration/custom-package-conf/packages/"
+        rmDir "test/integration/custom-package-conf/packages-extra/"
+        rmDir "test/integration/custom-package-conf/foo/dist/"
```

Comment:
Instead of having this additional test, which is relatively slow, I would prefer to have unit tests for the code that gets the package databases from the environment.  E.g. add

```haskell
getSandboxPackageDBsFromEnv :: IO [FilePath]
```
to the `Util` module and write tests for that.  `getSandboxPackageDBsFromEnv` would handle both `HASKELL_PACKAGE_SANDBOX` and `HASKELL_PACKAGE_SANDBOXES`.  And we would have four unit tests for that.